internal/btf/btf.go

```diff
@@ -454,8 +454,10 @@ func ProgramLineInfos(s *Program) (recordSize uint32, bytes []byte, err error) {
 // IsNotSupported returns true if the error indicates that the kernel
 // doesn't support BTF.
 func IsNotSupported(err error) bool {
-	ufe, ok := errors.Cause(err).(*internal.UnsupportedFeatureError)
-	return ok && ufe.Name == "BTF"
+	if xerrors.Is(err, (err).(*internal.UnsupportedFeatureError)) {
```

Comment:
I think this will panic if called on something that isn't an UFE.
```suggestion
	var ufe internal.UnsupportedFeatureError
	return xerrors.As(err, &ufe) && ufe.Name == "BTF"
```